Automattic/wp-calypso
# Domains: Remove old NameServers component

### Changes proposed in this Pull Request

After the redesigned domain management pages were deployed with #59241 and even after the cleanup in #60005, some obsolete code still remains. This PR removes the old `NameServers` component and related code that's not used anymore. Hat tip to @flootr for the nudge in [this comment](https://github.com/Automattic/wp-calypso/pull/59430#issuecomment-1090230757).

There were still some links to the old name servers management page which were replaced by links to the domain management page, where name servers are managed now. These are some of the places where these links appeared:

- Newly transferred domain notice
![Markup on 2022-04-08 at 09:44:40](https://user-images.githubusercontent.com/5324818/162448211-0b3c2c20-6cc8-459a-bfcb-18f84d9ad880.png)

- Domain deletion dialog
![Markup on 2022-04-08 at 09:49:59](https://user-images.githubusercontent.com/5324818/162448274-12ae5532-98cd-4251-af86-d64e63db8c82.png)

- Back link in mobile view when in the DNS management page
![Markup on 2022-04-08 at 10:19:34](https://user-images.githubusercontent.com/5324818/162448315-15e3c54d-c3f4-4466-b05a-9228d2a84b80.png)

### Testing instructions

- The changes here are simple enough that only code inspection should be ok.
- But please do a smoke test
  - Try to delete a domain and clicking the "move it to another service" link and confirm it takes you to the domain management page
  -  Go to the domain DNS management page in mobile view and ensure the back link takes you back to the domain management age
